File: Applied Course/4.Problem Solving/4.Problems on Stacks and Queues/5.Stock Span Problem Without using stack.py
def calculateSpan(price,n,stock):
    stock[0]=1
    for i in range(1,n):
        c=1
        while ((i-c)>=0) and (price[i] > price[i - c]):
            c += stock[i - c]
        stock[i]=c
    return stock
            


if __name__=='__main__':
    price=[10, 4, 5, 90, 120, 80]
    n=len(price)
    stock=[0]*n
    stock=calculateSpan(price,n,stock)
    for i in stock:
        print("{} ".format(i)),

File: Applied Course/4.Problem Solving/4.Problems on Stacks and Queues/5.Stock Span Problem Without using stack.c
#include <stdio.h>

void calculateSpan(int *arr, int n, int *stock) 
{ 
    // Span value of first element is always 1 
    stock[0] = 1; 
  
    //find stock values for rest 
    for (int i = 1; i < n; i++) { 
        int c = 1; 
        while ((i - c) >= 0 && arr[i] > arr[i - c]) { 
            c += stock[i - c]; 
        } 
        stock[i] = c; 
    } 
} 
  
//print elements of array 
void printSpan(int *arr, int n) 
{ 
    for (int i = 0; i < n; i++) 
        printf("%d\t", arr[i]); 
} 
  

int main() 
{ 
    int price[] = { 10, 4, 5, 90, 120, 80 }; 
    int n = sizeof(price) / sizeof(price[0]); 
    int stock[n]; 
    calculateSpan(price, n, stock); 
    printSpan(stock, n); 
  
    return 0; 
} 

